NeuroID/src/main/java/com/neuroid/tracker/utils/NIDComposeTextWatcherUtils.kt

```diff
@@ -0,0 +1,44 @@
+package com.neuroid.tracker.utils
+
+import com.neuroid.tracker.NeuroID
+
+class NIDComposeTextWatcherUtils(val neuroID: NeuroID) {
+    fun isPaste(
+        changeText: TextChange,
+        clipboardContent: String,
+        lastPastedHashCode: Int,
+    ): Boolean {
+        if (clipboardContent.isNotEmpty()) {
+            // send paste event here!
+            val pasteCount = clipboardContent.length
+            if (changeText.changedText.contains(clipboardContent) && (pasteCount == changeText.count)) {
+                // yeah we most likely have a paste event
+                val currentPastedHashCode = clipboardContent.hashCode()
+                // check for duplicate paste content
+                if (currentPastedHashCode != lastPastedHashCode) {
+                    // return true to indicate paste event
+                    return true
+                }
+            }
+        }
+        // no paste event if we get here
+        return false
+    }
+
+    fun getTextChange(
+        previousText: String,
+        currentText: String,
+    ): TextChange {
+        // equivalent of the TextWatcher, we need the count and changedText
+        val start = currentText.commonPrefixWith(previousText).length
```

Comment:
What happens when the current text change is less than the starting length. 

For example, when I run `getTextChange("zz", "")`

The result is: `TextChange(start = 0, count = 0, changedText = "")` Which doesnt seem. quite right (or I could me misunderstanding it)

Reply:
You delete two characters so the count of the left over characters is 0, the start is at 0 index and the changedText  is basically the deletions which is "". That is what I am seeing. I'll add a test case for deletions. For our case this is enough info to calculate pastes.